contracts/contracts/strategies/FraxETHStrategy.sol

```diff
@@ -0,0 +1,115 @@
+ * @notice Investment strategy for vaults supporting ERC4626
+ * @author Origin Protocol Inc
+ */
+import { IERC4626 } from "../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "../utils/InitializableAbstractStrategy.sol";
+import { IWETH9 } from "../interfaces/IWETH9.sol";
+import { IFraxETHMinter } from "../interfaces/IFraxETHMinter.sol";
+import { Generalized4626Strategy } from "./Generalized4626Strategy.sol";
+
+contract FraxETHStrategy is Generalized4626Strategy {
+    using SafeERC20 for IERC20;
+
+    IWETH9 public constant weth =
+        IWETH9(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+    IFraxETHMinter public constant fraxETHMinter =
+        IFraxETHMinter(0xbAFA44EFE7901E04E39Dad13167D089C559c1138);
+
+    function _deposit(address _asset, uint256 _amount) internal override {
+        require(_amount > 0, "Must deposit something");
+        require(
+            _asset == address(weth) || _asset == address(assetToken),
+            "Unexpected asset address"
+        );
+
+        if (_asset == address(weth)) {
+            // Unwrap WETH
+            weth.withdraw(_amount);
+
+            // Deposit ETH for frxETH
+            // slither-disable-next-line unused-return
+            fraxETHMinter.submitAndDeposit{ value: address(this).balance }(
+                address(this)
+            );
+        } else {
+            // Stake frxETH
+            // slither-disable-next-line unused-return
+            IERC4626(platformAddress).deposit(_amount, address(this));
+        }
+
+        emit Deposit(_asset, address(shareToken), _amount);
+    }
+
+    /**
+     * @dev Retuns bool indicating whether asset is supported by strategy
+     * @param _asset Address of the asset
+     */
+    function supportsAsset(address _asset)
+        external
+        view
+        override
+        returns (bool)
+    {
+        return _asset == address(assetToken) || _asset == address(weth);
+    }
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        virtual
+        override
+        returns (uint256 balance)
+    {
+        if (_asset == address(weth)) {
+            // For WETH, it's always 0
+            return 0;
+        }
+
+        // If it's not WETH, it has to be frxETH
+        require(_asset == address(assetToken), "Unexpected asset address");
+
+        /* We are intentionally not counting the amount of assetToken parked on the
+         * contract toward the checkBalance. The deposit and withdraw functions
+         * should not result in assetToken being unused and owned by this strategy
+         * contract.
+         */
+        return
+            IERC4626(platformAddress).convertToAssets(
```

Comment:
`convertToAssets` does not take into account fees or slippage. This probably does apply for sfrxETH, but for other vaults it does. I would recommend using `maxWithdraw` instead.

From the 4626 spec "The methods totalAssets, convertToShares and convertToAssets are estimates useful for display purposes, and do not have to confer the exact amount of underlying assets their context suggests."

Reply:
I agree with @naddison36 it would be more accurate for the `checkBalance` to report what is the actual amount extractable from the strategy. 

I've ran both the holdings of our strategy using `convertToAssets` resulting in `2013.0097359133865` and `maxWithdraw`  resulting in `2013.0097359133865`. The difference between the two values is `-0.00010626067205521394`. Interestingly the `maxWithdraw` gives a slightly higher number. 

From the [documentation](https://ethereum.org/en/developers/docs/standards/tokens/erc-4626/#maxwithdraw) it does look like as Nick states that using `maxwithdraw` would be a more accurate approach.